### PR TITLE
Add check in parallel config

### LINKED ISSUE #72: Proposal: allow running healtchecks in parallel
Currently the healthchecks are run in sequence:

    # Public: Run the registry's checks
    def run
      checks.each(&:run)
    end

this presents an issue for us as we have a constantly growing number of healthchecks but we also want to have a fixed known timeout for all healthchecks to be completed. It would be great to allow to run the healthchecks in parallel. What do you think?
I can create a pull request if you think this is a reasonable suggestion.

### PR DESCRIPTION
**Pull request by sergeyevstifeev**
Fixes #72

### CODE REVIEW

File: spec/ok_computer/check_collection_spec.rb [hunk 1]
```diff
@@ -2,59 +2,64 @@
 
 module OkComputer
   describe CheckCollection do
-    let(:foocheck) { double(:check) }
-    let(:barcheck) { double(:check) }
-    let(:registry) { {foo: foocheck, bar: barcheck} }
+    [false, true].each do |check_in_parallel|
```
Comment (pbyrne) [inline spec/ok_computer/check_collection_spec.rb]: I _think_ we'd only need to check both states of `OkComputer.check_in_parallel` in the `#run` test.

File: lib/ok_computer/check_collection.rb [hunk 2]
```diff
@@ -48,5 +48,19 @@ def to_json(*args)
...
     end
+
+    private
+
+    def check_in_sequence
+      checks.each(&:run)
+    end
+
+    def check_in_parallel
+      threads = []
```
Comment (pbyrne) [inline lib/ok_computer/check_collection.rb]: Minor nit-pick, but would something like this work?

```ruby
threads = checks.map do |check|
  Thread.new { check.run }
end
threads.each(&:join)
```

This option is probably _too_ terse, though:

```ruby
checks.map do |check|
  Thread.new { check.run }
end.each(&:join)
```

Comment (sergeyevstifeev) [inline lib/ok_computer/check_collection.rb]: The first option looks nice indeed. I've updated the code.

File: spec/ok_computer/check_collection_spec.rb (hunk 1 above)
Comment (sergeyevstifeev) [inline spec/ok_computer/check_collection_spec.rb]: Fixed

**Comment by pbyrne** [comment]
Looks good to me. Any feedback, @anfleene?

@sergeyevstifeev Barring further feedback, could you squash these commits down, and we'll get a release out.

**Comment by sergeyevstifeev** [comment]
Done

**Comment by anfleene** [comment]
Looks good :+1:

**Comment by pbyrne** [comment]
v1.6.0 is available on rubygems.org!

**Comment by sergeyevstifeev** [comment]
Awesome! :smiley:

**Pull request merged**